mono/utils/mono-jemalloc.h

```diff
@@ -0,0 +1,35 @@
+/**
+ * \file
+ *
+ * Header for jemalloc registration code
+ */
+
+#ifndef __MONO_JEMALLOC_H__
+#define __MONO_JEMALLOC_H__
+
+#if defined(MONO_JEMALLOC_ENABLED)
+
+/* Jemalloc can be configured in three ways.
+ * 1. You can use it with library loading hacks at run-time
+ * 2. You can use it as a global malloc replacement
+ * 3. You can use it with a prefix. If you use it with a prefix, you have to explicitly name the malloc function.
+ *
+ * In order to make this feature able to be toggled at run-time, I chose to use a prefix of mono_je. 
+ * This mapping is captured below in the header, in the spirit of "no magic constants".
+ *
+ */
+
+
+#include <jemalloc/jemalloc.h>
+
+#define MONO_JEMALLOC_MALLOC mono_jemalloc
+#define MONO_JEMALLOC_REALLOC mono_jerealloc
```

Comment:
Where are these functions defined ?

Reply:
The magic happens here: https://github.com/mono/mono/pull/5643/files#diff-64e89b1efb0df96de7d470b217d4a1a6R40

It would be nice to add to the comment above that the prefix is set in the makefile.

Reply:
That comment is on line 13?

Reply:
```  * In order to make this feature able to be toggled at run-time, I chose to use a prefix of mono_je. 
```

I can clarify that it's set in the Makefile, rather than just saying that jemalloc is "configured" that way. I was thinking it was clear that I meant autoconf.